dags/common/utils.py

```diff
@@ -228,3 +227,61 @@ def process_archive(file_bytes, file_name):
         return process_zip_file(file_bytes, file_name)
     if tarfile.is_tarfile(file_bytes):
         return process_tar_file(file_bytes, file_name)
+
+
+def process_zip_file_elsevier(file_bytes, file_name):
+    file_bytes.seek(0)
+    with zipfile.ZipFile(file_bytes) as zip:
+        for filename in zip.namelist():
+            if "dataset.xml" in filename:
+                file_prefix = ".".join(file_name.split(".")[:-1])
+                zip_file_content = zip.read(filename)
+                s3_filename = os.path.join(file_prefix, filename)
+                return (zip_file_content, s3_filename)
```

Comment:
I would use the following pattern in order to avoid nested branches :)

```suggestion
        for filename in zip.namelist():
            if "dataset.xml" not in filename:
                continue
            file_prefix = ".".join(file_name.split(".")[:-1])
            zip_file_content = zip.read(filename)
            s3_filename = os.path.join(file_prefix, filename)
            return (zip_file_content, s3_filename)
```